HaruhiGekidouTests/SourceTests.cs

```diff
@@ -44,6 +47,52 @@ public class SourceTests
             "Tutorial_006"
         ];
 
+        public static string[] AdvPartScriptFiles()
+        {
+            if (Directory.Exists(Path.Combine("input", "AdvPartScript")))
+            {
+                return Directory.GetFiles(Path.Combine("input", "AdvPartScript"));
+            }
+            else
+            {
+                Directory.CreateDirectory(Path.Combine("input", "AdvPartScript"));
+
+                List<string> newScripts = new List<string>();
+                string [] arcs = Directory.GetFiles(Path.Combine("input", "ScriptArc"));
+                foreach (string arc in arcs)
+                {
+                    if (_scriptArcNames.Contains(Path.GetFileNameWithoutExtension(arc)))
+                    {
+                        byte[] arcBytes = File.ReadAllBytes(arc);
+                        GekidouArc newArc = new(arcBytes);
+                        
+                        bool isInScript = false;
+                        foreach (GekidouArcEntry entry in newArc.Entries)
+                        {
+                            if (entry.IsDirectory)
+                            {
+                                isInScript = entry.Name == "AdvPartScript";
+                            }
+                            else
+                            {
+                                if (!isInScript) continue;
+                                string name = Path.GetFileNameWithoutExtension(entry.Name);
+                                if (arc.Contains("Tutorial"))
+                                {
+                                    name += arc.Split("_")[1];
+                                    
+                                }
+                                File.WriteAllBytes(Path.Combine(Path.Combine("input", "AdvPartScript"), name + ".bin"), entry.Data);
```

Comment:
No need to nest the Path.Combines here:

```suggestion
                                File.WriteAllBytes(Path.Combine("input", "AdvPartScript", name + ".bin"), entry.Data);
```

Reply:
Also I'd use an interpolated string:
```cs
$"{name}.bin"
```